lib/src/components/NxCheckbox/__tests__/NxCheckbox.test.tsx

```diff
@@ -145,10 +146,25 @@ describe('NxCheckbox', function() {
     expect(component).toHaveProp('htmlFor', 'baz');
   });
 
-  it('pass attributes into checkbox input element', function() {
-    expect(getShallowComponent({ checkboxAttributes: { name: 'Garfield'} })
-        .find('input')).toHaveProp('name', 'Garfield');
-    expect(getShallowComponent({ checkboxAttributes: { className: 'cat', name: 'Garfield'} })
-        .find('input')).toHaveClassName('cat');
+  it('passes input attributes into the input element and does not clash with top-level attributes', function() {
+    const component = getShallowComponent({
+      checkboxId: 'not-garfield',
+      disabled: true,
+      isChecked: true,
+      className: 'label-classname',
+      inputAttributes: {
+        id: 'garfield',
+        name: 'garfield',
+        disabled: false,
+        className: 'checkbox-classname',
+        checked: false
+      }
+    });
+
+    expect(component.find('input')).toHaveProp('id', 'garfield');
+    expect(component.find('input')).toHaveProp('name', 'garfield');
+    expect(component.find('input')).toHaveProp('disabled', true);
+    expect(component.find('input')).toHaveClassName('checkbox-classname');
```

Comment:
Wouldn't hurt to also check that the `input` doesn't have `label-classname`

Reply:
> Wouldn't hurt to also check that the `input` doesn't have `label-classname`

Agreed! Added test in this commit: d9cf5af435f57153f63578b8eec1a151a520a8cc